odoo/tools/__init__.py

```diff
@@ -18,3 +19,17 @@
 from .debugger import *
 from .xml_utils import *
 from .date_utils import *
+
+def json_default(obj):
+    """
+    Properly serializes date and datetime objects.
+    """
+    from odoo import fields
+    if isinstance(obj, datetime):
+        return fields.Datetime.to_string(obj)
+    if isinstance(obj, date):
+        return fields.Date.to_string(obj)
+    if isinstance(obj, lazy):
+        return obj._value
+    return ustr(obj)
+date_utils.json_default = json_default
```

Comment:
I wonder whether this is still relevant: https://github.com/odoo/odoo/blob/2a2ff6eb5c746acec46ebc7fee382dfa78ec8c14/odoo/tools/func.py#L247-L255

Reply:
Might be for arbitrary / random json.dumps I guess? Unclear whether there's a risk those would get a lazy coming their way.

Reply:
Tried to remove it, was only used (nearly explicitly) in a test, so I updated the test: https://github.com/odoo/odoo/blob/307be80ec659c6c2d1e2371d2eb01d345fbb275f/odoo/addons/test_performance/tests/test_performance.py#L142-L144

Reply:
> Tried to remove it, was only used (nearly explicitly) in a test, so I updated the test:

Why would you change this in a stable version? Was this part necessary to fix the bug?